src/Microsoft.Data/DataFrame.cs

```diff
@@ -133,13 +133,124 @@ public DataFrame Sort(string columnName, bool ascending = true)
             for (int i = 0; i < ColumnCount; i++)
             {
                 BaseColumn oldColumn = Column(i);
-                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(sortIndices, !ascending);
+                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(oldColumn.NullCount, sortIndices, !ascending);
                 Debug.Assert(newColumn.NullCount == oldColumn.NullCount);
                 newColumns.Add(newColumn);
             }
             return new DataFrame(newColumns);
         }
 
+        public enum JoinAlgorithm
+        {
+            Left,
+            Right,
+            Outer,
+            Inner
+        }
+
+        public DataFrame Join(DataFrame other, string leftSuffix = "_left", string rightSuffix = "_right", JoinAlgorithm joinAlgorithm = JoinAlgorithm.Left)
```

Comment:
Don't you need something to join "on"? For example, Join on `left.CustomerId == right.Id`.

Reply:
Nope. This is a straight index on index join in Pandas speak. Maybe you are thinking of `Merge`?

Reply:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.join.html

> on : str, list of str, or array-like, optional
Column or index level name(s) in the caller to join on the index in other, otherwise joins index-on-index. If multiple values given, the other DataFrame must have a MultiIndex. Can pass an array as the join key if it is not already contained in the calling DataFrame. Like an Excel VLOOKUP operation.

Reply:
Right, I should've explained better. So, we can't support `on` yet because the `DataFrame`  does not understand `index` yet. In Pandas, each dataframe has an index associated with it. So I can do this:
`df = pd.DataFrame({'key': ['K0', 'K1', 'K2', 'K3', 'K4', 'K5'],
...                    'A': ['A0', 'A1', 'A2', 'A3', 'A4',
other = pd.DataFrame({'key': ['K0', 'K3', 'K5'],
...                       'B': ['B0', 'B3', 'B5']})
df.join(other.set_index('key'), on='key')`
where it would use the `key` column in `other` as its index. For our prototype here(as a v1.0), I thought using the row numbers as an implicit index is good enough? Thoughts?

Reply:
If we need to support `index` before implementing `on`, then let's continue as-is for now without the `on` support.